specifyweb/frontend/js_src/lib/components/Formatters/Element.tsx

```diff
@@ -91,10 +99,19 @@ export function XmlEditorShell<
         <Label.Block>
           {resourcesText.name()}
           <Input.Text
+            forwardRef={validationRef}
             isReadOnly={isReadOnly}
             required
             value={item.name}
-            onValueChange={(name): void => setItem({ ...item, name })}
+            onValueChange={(name): void => {
+              setItem({ ...item, name });
+              setValidation(
+                uniqueNames.includes(name)
+                  ? resourcesText.duplicateFormatters()
+                  : '',
+                'focus'
+              );
```

Comment:
is this needed? 

isn't this code you added higher up already doing this?
```ts
hasDuplicates ? resourcesText.duplicateFormatters() : ''
```

Reply:
yes you're write I might have to added it because I had my 'save' button disabled